### PR TITLE
348 - feat: create payment

### PR DESCRIPTION
**Pull request by emmaacha**
Ticket IDs: 
- [kh-books: #348: [PAYMENTS] Create Payment](https://github.com/nerubia/kh-books/issues/348)
- [kh-books: #356: [PAYMENTS] Generate Payment No. #356](https://github.com/nerubia/kh-books/issues/356)

Related kh-books PR: [348 - feat: create payment #385](https://github.com/nerubia/kh-books/pull/385)

### CODE REVIEW

File: src/repositories/khbooks/payment-repository.ts [hunk 1]
```diff
@@ -39,3 +39,107 @@ export const countAllByFilters = async (where: Prisma.paymentsWhereInput) => {
...
+          },
+        ],
+      },
+      orderBy: {
+        payment_no: "desc",
+      },
+      take: 1,
+    })
+
+    let paymentNo = Number(process.env.INVOICE_NO_OFFSET ?? 0)
```
Comment (jatnerubia) [inline src/repositories/khbooks/payment-repository.ts:77]: Payment number should only be 4 digits. Please create a separate env variable for this ^^

File: src/repositories/khbooks/payment-repository.ts [hunk 2]
```diff
@@ -39,3 +39,107 @@ export const countAllByFilters = async (where: Prisma.paymentsWhereInput) => {
...
+
+    if (payments.length === 1) {
+      paymentNo = Number(payments[0].payment_no) + 1
+    }
+
+    const formattedPaymentNo = paymentNo.toString().padStart(6, "0")
```
Comment (jatnerubia) [inline src/repositories/khbooks/payment-repository.ts:83]: ^^

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

**Review by jatnerubia — APPROVED** [review]
Cool

File: src/repositories/khbooks/payment-repository.ts (hunk 1 above)
Comment (emmaacha) [reply src/repositories/khbooks/payment-repository.ts]: I think no need for offset, since it will start at `0000` according to the ticket

**Pull request merged**